Fix a flaky integration test

https://circleci.com/gh/neuromation/platform-storage-api/268

```
    @pytest.mark.asyncio
    async def test_alice_rename_file_to_bobs_relative(
        self, server_url, api, client, alice, bob
    ):
        await self.put_file(server_url, client, alice, self.file1, self.payload1)
        status = await self.get_filestatus(server_url, client, alice, alice, self.file1)
        response = await self.rename_relative(
            server_url, client, alice, alice, self.file1, f"../{bob.name}/self.file2"
        )
>       assert response.status == aiohttp.web.HTTPNotFound.status_code
E       AssertionError: assert 500 == 404
E        +  where 500 = <ClientResponse(http://storage:5000/api/v1/storage/54fe5180-71e8-426a-a042-db21ac59c505/file1?op=RENAME&destination=.....arset=utf-8', 'Content-Length': '221', 'Date': 'Sat, 22 Dec 2018 14:36:50 GMT', 'Server': 'Python/3.6 aiohttp/3.4.4')>\n.status
E        +  and   404 = <class 'aiohttp.web_exceptions.HTTPNotFound'>.status_code
E        +    where <class 'aiohttp.web_exceptions.HTTPNotFound'> = <module 'aiohttp.web' from '/usr/local/lib/python3.6/site-packages/aiohttp/web.py'>.HTTPNotFound
E        +      where <module 'aiohttp.web' from '/usr/local/lib/python3.6/site-packages/aiohttp/web.py'> = aiohttp.web

tests/integration/test_api.py:819: AssertionError
```

--- a/platform_storage_api/api.py
+++ b/platform_storage_api/api.py
@@ -270,6 +270,7 @@ class StorageHandler:
             new = PurePath(request.query["destination"])
             if new.root == "":
                 new = old.parent / new
+            new = self._storage.sanitize_path(new)
             await self._check_user_permissions(request, str(new))
             await self._storage.rename(old, new)
         except FileNotFoundError:

